raimitigations/dataprocessing/imputer/knn_imputer.py

```diff
@@ -0,0 +1,262 @@
+from typing import Union
+import pandas as pd
+import numpy as np
+import sys
+
+from sklearn.impute import KNNImputer
+
+from ..encoder import EncoderOrdinal
+from .imputer import DataImputer
+from ...utils.data_utils import get_cat_cols
+
+import warnings
+
+
+def custom_formatwarning(msg, *args, **kwargs):
+    # ignore everything except the message
+    return str(msg) + "\n"
+
+
+warnings.formatwarning = custom_formatwarning
+
+
+class KNNDataImputer(DataImputer):
+    """
+    Concrete class that imputes missing data of a feature using K-nearest neighbours. A feature's missing values are imputed using the mean value from k-nearest neighbors in the dataset. Two samples are close if the features that neither is missing are close.
+    This subclass uses the :class:`~sklearn.impute.KNNImputer` class from :mod:`sklearn` in the background.
+    sklearn.impute.KNNImputer can only handle numerical data, however, this subclass allows for categorical input by applying ordinal encoding before calling the sklearn class. In order to use this function, use enable_encoder=True, note that encoded columns are not guaranteed to reverse transform if they have imputed values.
```

Comment:
I think there should be a period here:

- "...function, use enable_encoder=True, note that..."
- "...function, use enable_encoder=True. Note that..."